### ISSUE TITLE
modify internal URL as my_app.worker.1 is invalid url due to '.1' at the end

### ISSUE THREAD

**Issue by lmdc45** [dokku/dokku#6993]
### Description of feature

I tried using 'Specifying a custom TLD' but that seems not working within an app?
I tried setting it to .local so I could use my_app.worker.1.local, but still appears as my_app.worker.1

issue appears after switching from nodev14 to nodev22

new URL('http://my_app.worker.1') =>error 
new URL('http://my_app.worker.1.local') =>would work fine

**Comment by josegonzalez** [comment]
That sounds like a nodejs error, not a Dokku error. I won't be changing how we generate the TLD.

You can alternatively not address a specific instance of the app by doing `http://app-name.worker` (omitting the index) which is also added as an alias to each instance of that process type.

**Comment by lmdc45** [comment]
FYI below, it seems not a nodejs/javascript issue but maybe the standard that should not apply to 'internal' url
------
The WHATWG URL Standard is a specification developed by the Web Hypertext Application Technology Working Group (WHATWG) that defines how URLs (Uniform Resource Locators) should be parsed, formatted, and processed. This standard ensures consistency across different browsers and platforms, aiming to provide a clear and uniform way of handling URLs.

To determine if the URL `http://app.web.1` complies with the WHATWG URL Standard, let's break it down:

1. **Scheme**: `http` - This is valid.
2. **Host**: `app.web.1` - According to the standard, the host can be a domain name, an IPv4 address, or an IPv6 address. However, domain names should follow specific rules, such as not ending with a digit and having valid top-level domains (TLDs).

Here, `app.web.1` doesn't conform to the typical domain name structure, primarily because:
- TLDs are usually alphabetic (e.g., `.com`, `.org`, `.net`) and `.1` is not a recognized TLD.
- Having a numerical TLD (`.1`) is unconventional and doesn't adhere to standard domain naming practices.

Therefore, the URL `http://app.web.1` does not comply with the WHATWG URL Standard because the domain name structure is invalid under this specification.

For internal URLs within apps, such as those used in local testing environments or within internal networks, compliance with the WHATWG URL Standard can be more flexible. However, it’s still a good practice to follow standard URL conventions to avoid potential issues.

-----

I tried using app.worker but it does not resolve, only app.worker.1 does. Noting the app only have one web and one worker process.

I tried setting the network tld to .local, however same app.worker.1.local seems not resolving 

I workaround by resolving the ip first, which works fine with dns.resolve within nodejs, then build the url from it
